### PR TITLE
feat : Leave type with partial payment

### PR DESCRIPTION
**Pull request by Anurag810**
image

Docs: https://github.com/frappe/erpnext_documentation/pull/135

### CODE REVIEW

**Comment by Anurag810** [comment]
docs: https://github.com/frappe/erpnext_documentation/pull/135

File: erpnext/payroll/doctype/salary_slip/salary_slip.py [hunk 1]
```diff
@@ -328,19 +330,36 @@ def calculate_lwp_based_on_leave_application(self, holidays, working_days):
...
-	def calculate_lwp_and_absent_days_based_on_attendance(self, holidays):
+	def calculate_lwp_ppl_and_absent_days_based_on_attendance(self, holidays):
 		lwp = 0
 		absent = 0
 
 		daily_wages_fraction_for_half_day = \
 			flt(frappe.db.get_value("Payroll Settings", None, "daily_wages_fraction_for_half_day")) or 0.5
 
-		lwp_leave_types = dict(frappe.get_all("Leave Type", {"is_lwp": 1}, ["name", "include_holiday"], as_list=1))
+		leave_types = frappe.db.sql(""" Select name, is_lwp, is_ppl, fraction_of_daily_salary_per_leave, include_holiday
```
Comment (rmehta) [inline erpnext/payroll/doctype/salary_slip/salary_slip.py]: You also have `or_conditions` in get_all

**Review by rmehta — CHANGES_REQUESTED** [review]
test missing

**Comment by Anurag810** [comment]
There already a test for LWP and a fraction of PPL is added in LWP.

**Comment by sagarvora** [comment]
@Anurag810 Please fix breaking checks.

File: erpnext/payroll/doctype/salary_slip/salary_slip.js [hunk 2]
```diff
@@ -123,8 +123,8 @@ frappe.ui.form.on("Salary Slip", {
 			doc: frm.doc,
 			callback: function(r, rt) {
 				frm.refresh();
-				if (r.message){
-					frm.fields_dict.absent_days.set_description("Unmarked Days is treated as "+ r.message +". You can can change this in " + frappe.utils.get_form_link("Payroll Settings", "Payroll Settings", true));
+				if (r.message[1] !== "Leave" && r.message[0]){
+					frm.fields_dict.absent_days.set_description("Unmarked Days is treated as "+ r.message[0] +". You can can change this in " + frappe.utils.get_form_link("Payroll Settings", "Payroll Settings", true));
```
Comment (nabinhait) [inline erpnext/payroll/doctype/salary_slip/salary_slip.js]: Make it multilingual

File: erpnext/hr/doctype/leave_type/leave_type.py [hunk 3]
```diff
@@ -21,3 +21,9 @@ def validate(self):
 			leave_allocation = [l['name'] for l in leave_allocation]
 			if leave_allocation:
 				frappe.throw(_('Leave application is linked with leave allocations {0}. Leave application cannot be set as leave without pay').format(", ".join(leave_allocation))) #nosec
+
+		if self.is_lwp and self.is_ppl:
+			frappe.throw(_("Leave Type can be either without pay or partial pay"))
+
+		if self.is_ppl and 0 < self.fraction_of_daily_salary_per_leave < 1 :
```
Comment (nabinhait) [inline erpnext/hr/doctype/leave_type/leave_type.py]:
```suggestion
		if self.is_ppl and (self.fraction_of_daily_salary_per_leave < 0 or  self.fraction_of_daily_salary_per_leave > 1):
```

**Review by nabinhait — CHANGES_REQUESTED** [review]
Please write a specific test case for partial paid leave

**Review by nabinhait — APPROVED** [review]

**Pull request merged**